src/providers/WorkspaceSymbolProvider.ts

```diff
@@ -1,85 +1,91 @@
 import * as vscode from "vscode";
 import { AtelierAPI } from "../api";
-import { ClassDefinition } from "../utils/classDefinition";
+import { currentWorkspaceFolder } from "../utils";
 import { DocumentContentProvider } from "./DocumentContentProvider";
-import { config } from "../extension";
 
 export class WorkspaceSymbolProvider implements vscode.WorkspaceSymbolProvider {
-  public provideWorkspaceSymbols(
-    query: string,
-    token: vscode.CancellationToken
-  ): vscode.ProviderResult<vscode.SymbolInformation[]> {
-    if (query.length < 3) {
+  private sql: string =
+    "SELECT * FROM (" +
+    "SELECT Name, Parent->ID AS Parent, 'method' AS Type FROM %Dictionary.MethodDefinition" +
+    " UNION ALL " +
```

Comment:
```suggestion
    " UNION ALL  %PARALLEL" +
```
And for all the next `UNION ALL`'s, it'll the whole query much faster.